pkg/dbal/DbalConsumerHelperTrait.php

```diff
@@ -105,9 +107,13 @@ protected function redeliverMessages(): void
             ->setParameter('redelivered', true, Type::BOOLEAN)
         ;
 
-        $update->execute();
+        try {
+            $update->execute();
 
-        $this->redeliverMessagesLastExecutedAt = microtime(true);
+            $this->redeliverMessagesLastExecutedAt = microtime(true);
+        } catch (DeadlockException $e) {
```

Comment:
it would be better to use `\Doctrine\DBAL\Exception\RetryableException` that I created. this would also cover lock timeouts.